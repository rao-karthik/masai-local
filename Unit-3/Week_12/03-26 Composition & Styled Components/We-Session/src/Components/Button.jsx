import React from 'react';
import styles from './Button.module.css'

export const Button = ({children, onClick}) => {
    return (
        <div className = {styles.button} onCLick = {onClick}>
            {children}
        </div>
    )
}
