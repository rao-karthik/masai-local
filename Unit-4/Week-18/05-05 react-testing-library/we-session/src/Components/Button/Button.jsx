import React from 'react'

export const Button = ({ label, onClick, color }) => {

    const style = { backgroundColor: color }
    
    return (
        <button data-testid="button-form" style={style} onClick={onClick}>
            {label}
        </button>
    )
}
