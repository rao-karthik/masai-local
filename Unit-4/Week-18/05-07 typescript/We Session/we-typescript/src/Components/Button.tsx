import React from 'react';

interface IButtonProps {
    label: string;
    handleClick: ()=>void
}

export const Button = ({label, handleClick}: IButtonProps) => {
    return (
        <button onClick={handleClick}>
            {label}
        </button>
    )
}
