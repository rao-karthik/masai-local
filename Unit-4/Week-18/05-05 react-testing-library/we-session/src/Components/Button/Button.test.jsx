import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { Button } from './Button';
import renderer from 'react-test-renderer';

describe("Button component Testing",()=>{
    const buttonTestId = "button-form";

    // * Test the component itself is failing

    it("should render the label correctly", ()=>{
        const label = "TESTING";
        const { getByTestId } = render(<Button label={label} />)

        // 

        const btn = getByTestId(buttonTestId);
        expect(btn.textContent).toBe(label)
    })

    // * functionality

    it("button should invoke the onClick prop", ()=>{
        const mockFn = jest.fn();
        const label = "ADD";
        render( <Button label={label} onClick={mockFn} />)

        const btn = screen.getByTestId(buttonTestId);

        fireEvent.click(btn);
        fireEvent.click(btn);
        fireEvent.click(btn);
        fireEvent.click(btn);
        fireEvent.click(btn);

        expect(mockFn).toHaveBeenCalled();
        expect(mockFn).toHaveBeenCalledTimes(5)
    })
    
    // * button colors

    it("Button colors should work correctly", ()=>{
        const label = "COLOR";
        const color = "red";
        render (<Button label={label} color={color} />)

        const btn = screen.getByTestId(buttonTestId);

        expect(btn.textContent).toBe(label);
        expect(btn.style.backgroundColor).toBe(color);
    })

    //* snapshot

    it("matches snapshot", ()=>{
        const tree = renderer.create(<Button color="green" label="snap" />)
        expect(tree).toMatchSnapshot();
    })
})