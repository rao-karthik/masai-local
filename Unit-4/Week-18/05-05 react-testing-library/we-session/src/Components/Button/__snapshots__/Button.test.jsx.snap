// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Testing matches snapshot 1`] = `
<button
  data-testid="button-form"
  style={
    Object {
      "backgroundColor": "green",
    }
  }
>
  snap
</button>
`;
